src/node_dir.cc

```diff
@@ -160,7 +160,69 @@ void DirHandle::Close(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
-void AfterDirReadSingle(uv_fs_t* req) {
+static MaybeLocal<Array> DirentListToArray(
+    Environment* env,
+    uv_dirent_t* ents,
+    int num,
+    enum encoding encoding,
+    Local<Value>* err_out) {
+  MaybeStackBuffer<Local<Value>, 96> entries(num * 3);
+
+  // Return an array of all read filenames.
+  int j = 0;
+  for (int i = 0; i < num; i++) {
+    Local<Value> filename;
+    Local<Value> error;
+    const size_t namelen = strlen(ents[i].name);
+    if (!StringBytes::Encode(env->isolate(),
+                             ents[i].name,
+                             namelen,
+                             encoding,
+                             &error).ToLocal(&filename)) {
+      *err_out = error;
+      return MaybeLocal<Array>();
+    }
+
+    entries[j++] = filename;
+
+    // We cache the returned values in JS. However, the API allows multiple
+    // read() calls with different `encoding` options; while this may be
```

Comment:
Oh god, should we just revert that before it gets released?

Reply:
I’d be okay with that too.